sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/FeedResponseDiagnostics.java

```diff
@@ -5,23 +5,31 @@
 
 import com.azure.cosmos.implementation.query.QueryInfo;
 import com.azure.cosmos.implementation.query.metrics.QueryMetricsTextWriter;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
 /**
  * The type Feed response diagnostics.
  */
-public final class FeedResponseDiagnostics {
+public class FeedResponseDiagnostics {
 
     private final static String EQUALS = "=";
     private final static String QUERY_PLAN = "QueryPlan";
     private final static String SPACE = " ";
+    private static final Logger LOGGER = LoggerFactory.getLogger(FeedResponseDiagnostics.class);
     private Map<String, QueryMetrics> queryMetricsMap;
     private QueryInfo.QueryPlanDiagnosticsContext diagnosticsContext;
+    private List<ClientSideRequestStatistics> clientSideRequestStatisticsList;
```

Comment:
Do we need this list be thread safe, multiple threads could update the diagnostics from multiple responses ?

Reply:
I dont see a specific case of this getting into thread related issues yet. Also we are just doing an add operation and generally thread related issues can be seen when we are iterating or add/remove operations. I would move it to a sync list if I come across any issues.

Reply:
Are we planning to run perf , because we cant be sure till we run some load. Without 100% surety it will be risky, as we have seen similar issue around diagnostics lists.

Reply:
Just for the record, the querymetricsMap thats currently being used in FeedResponseDiagnostics already is just a Map and not thread safe and did'nt lead to any issues and IMHO we would have run perf tests already on that?
Either ways I would run perf test on this anyway for the perf reasons, and that would expose these and will fix accordingly

Reply:
Tried multiple tests but didnt encounter thread issues on the list. But due to some known issues around existing lists which were eventually converted to be synchronized, updated this list to be synchronized.